geest/core/workflows/default_index_score_workflow.py

```diff
@@ -36,212 +35,110 @@ def __init__(
-        temp_layer_data_provider.addFeature(temp_feature)
-
-        # Ensure resolution parameters are properly formatted as float values
-        x_res = 100.0  # 100m pixel size in X direction
-        y_res = 100.0  # 100m pixel size in Y direction
-        aligned_box = aligned_box.boundingBox()
-        # Define rasterization parameters for the temporary layer
-        params = {
-            "INPUT": temp_layer,
-            "FIELD": None,
-            "BURN": index_score,  # todo Jeff put on likert scale properly
-            "USE_Z": False,
-            "UNITS": 1,
-            "WIDTH": x_res,
-            "HEIGHT": y_res,
-            "EXTENT": f"{aligned_box.xMinimum()},{aligned_box.xMaximum()},"
-            f"{aligned_box.yMinimum()},{aligned_box.yMaximum()} [{self.target_crs.authid()}]",  # Extent of the aligned bbox
-            "NODATA": None,
-            "OPTIONS": "",
-            "DATA_TYPE": 0,  # byte
-            "INIT": None,
-            "INVERT": False,
-            "EXTRA": "",
-            "OUTPUT": "TEMPORARY_OUTPUT",
-        }
-        # Run the rasterize algorithm
-        raster = processing.run("gdal:rasterize", params)["OUTPUT"]
+        :return: True if the workflow completes successfully, False if canceled or failed.
+        """
         QgsMessageLog.logMessage(
-            f"Created raster: {raster}", tag="Geest", level=Qgis.Info
+            f"Processing area {index} score workflow", "Geest", Qgis.Info
         )
+        output_prefix = f"{self.layer_id}_area_features_{index}"
 
-        # Clip the raster to the study area boundary
-        clipped_raster_filepath = os.path.join(
-            self.workflow_directory, f"{raster_filepath}"
+        QgsMessageLog.logMessage(
+            f"Index score: {self.index_score}",
+            "Geest",
+            Qgis.Info,
         )
 
-        processing.run(
-            "gdal:cliprasterbymasklayer",
-            {
-                "INPUT": raster,
-                "MASK": self.areas_layer,
-                "NODATA": 255,
-                "CROP_TO_CUTLINE": True,
-                "OUTPUT": raster_filepath,
-            },
+        # Create a scored boundary layer filtered by current_area
+        scored_layer = self.create_scored_boundary_layer(output_prefix)
+
+        # Create a scored boundary layer
+        raster_output = self._rasterize(
+            scored_layer,
+            current_bbox,
+            index,
+            value_field="score",
+            default_value=255,
         )
+        QgsMessageLog.logMessage(f"Raster output: {raster_output}", "Geest", Qgis.Info)
         QgsMessageLog.logMessage(
-            f"Created raster: {raster_filepath}", tag="Geest", level=Qgis.Info
+            f"Worflow completed for area {index}", "Geest", Qgis.Info
         )
+        return raster_output
 
-    def create_raster_vrt(self, output_vrt_name: str = None) -> None:
+    def create_scored_boundary_layer(self, output_prefix: str) -> QgsVectorLayer:
```

Comment:
Better to calculate the output prefix in the function since it is the only place it it's ever used